examples/lines_bars_and_markers/barh_demo.py

```diff
@@ -6,16 +6,20 @@
 
 
 plt.rcdefaults()
+fig, ax = plt.subplots()
 
 # Example data
 people = ('Tom', 'Dick', 'Harry', 'Slim', 'Jim')
 y_pos = np.arange(len(people))
 performance = 3 + 10 * np.random.rand(len(people))
 error = np.random.rand(len(people))
 
-plt.barh(y_pos, performance, xerr=error, align='center')
-plt.yticks(y_pos, people)
-plt.xlabel('Performance')
-plt.title('How fast do you want to go today?')
+ax.barh(y_pos, performance, xerr=error, align='center',
```

Comment:
If we do not want to backport this to v2.x, this can use @story645 's categorical work.  `ax.barh(people, performance)` should 'just work'